src/AElf.Kernel.Core/Blockchain/Domain/IChainManager.cs

```diff
@@ -410,5 +411,16 @@ private async Task RemoveChainBranchesAsync(Chain chain, List<string> branchKeys
 
             await _chains.SetAsync(chain.Id.ToStorageKey(), chain);
         }
+
+        public async Task RemoveCannotExecutedLongestBranchAsync(Chain chain)
```

Comment:
Can you name it in a OOP way ???

Reply:
How about "RemoveNonExecutableLongestBranchAsync"?

Reply:
> Can you name it in a OOP way ???

Sorry, I got it. Maybe "RemoveLongestBranchAsync" is better?